### PR TITLE
Add default values to form

### PR DESCRIPTION
**Pull request by Netznarkose**

### CODE REVIEW

File: app/views/charts/_form.html.erb [hunk 1]
```diff
@@ -14,22 +14,32 @@
...
+        <%= f.input(:item_height, input_html: { value: f.object.item_height || 20 }) %>
+        <%= f.input(:item_width, input_html: { value: f.object.item_width || 20 }) %>
+        <%= f.input(:width, input_html: { value: f.object.width || 1000 }) %>
+        <%= f.input(:height, input_html: { value: f.object.height || 1000 }) %>
         <%= f.input(:grouplabels) %>
-        <%= f.input(:item_height) %>
-        <%= f.input(:item_width) %>
-        <%= f.input(:file_type, collection: [['PNG', :png], ['SVG', :svg]], prompt: true) %>
-        <%= f.input(:width) %>
-        <%= f.input(:height) %>
```
Comment (Netznarkose) [inline app/views/charts/_form.html.erb]: any idea who to articulate this less redundant?

File: app/views/charts/_form.html.erb [hunk 2]
```diff
@@ -14,22 +14,32 @@
...
+        <% if @chart.persisted? %>
+          <%= f.input(:file_type, collection: [['PNG', :png], ['SVG', :svg]]) %>
```
Comment (Netznarkose) [inline app/views/charts/_form.html.erb]: any idea who to articulate this less redundant?

**Comment by eikes** [comment]
It is a better idea to set these default values in a model method which is called by the after_initialize hook:

http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html

Just make sure not override the existing value:

http://stackoverflow.com/a/16392285/334314

File: app/views/charts/_form.html.erb (hunk 2 above)
Comment (Netznarkose) [inline app/views/charts/_form.html.erb]: can you tell me what I do wrong with this approach?
```ruby
  after_initialize :vanilla_params, :if => :new_record?
  def vanilla_params
    {
      background_color: 'white'
    }
  end
```

Comment (eikes) [inline app/views/charts/_form.html.erb]: You'll need to assign the parameters to self. It's just a regular function. Your function only returns a hash, it has no effect on the model instance.

```ruby
  after_initialize :vanilla_params, if: :new_record?
  def vanilla_params
    self.background_color = 'white'
  end
```

Comment (Netznarkose) [inline app/views/charts/_form.html.erb]: I pushed a new version which implements default values this way!

**Pull request merged**